# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/AudioData.cpp
# Commit: c41a5aac3192
# Full Hash: c41a5aac3192352c9b19dd3fda35f6e5ee4a74ba
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-03-06 21:16:27
# Regressor Bug: 1858958
# File Overlap Count: 1
# Description:
#   Bug 1858958 - Test more conversions, clearly mark planar->interleaved conversion as outside of spec. r=chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D192339
# ==============================================================================

diff -r db5ccbbdbbee -r c41a5aac3192 dom/media/webcodecs/AudioData.cpp
--- a/dom/media/webcodecs/AudioData.cpp	Wed Mar 06 11:14:19 2024 +0000
+++ b/dom/media/webcodecs/AudioData.cpp	Wed Mar 06 11:14:19 2024 +0000
@@ -443,6 +443,7 @@
   }
 
   if (!IsInterleaved(aSourceFormat) && IsInterleaved(aCopyToSpec.mFormat)) {
+    MOZ_CRASH("This should never be hit -- current spec doesn't support it");
     // Planar to interleaved -- copy of all channels of the source into the
     // destination buffer.
     MOZ_ASSERT(aCopyToSpec.mPlaneIndex == 0);